### PR TITLE
[IMP] *: translations improvements

### PR DESCRIPTION
**Pull request by Feyensv**
--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale_coupon/models/sale_order.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def _get_reward_values_product(self, program):
             'price_unit': - price_unit,
             'product_uom_qty': reward_qty,
             'is_reward_line': True,
-            'name': _("Free Product") + " - " + program.reward_product_id.name,
+            'name': _("Free Product - %s", program.reward_product_id.name),
```
Comment (mart-e) [inline addons/sale_coupon/models/sale_order.py:105]: same here, only adding more complexity for the translators

Comment (Feyensv) [reply addons/sale_coupon/models/sale_order.py:105]: I'm maybe wrong, but in some languages, shouldn't the product name be placed before the "Free product" part of the text ? :thinking:

Comment (mart-e) [reply addons/sale_coupon/models/sale_order.py:105]: The LTR is a rendering matter, you can display the same arabic text in LTR and RTL (doesn't mean anything but that's what you got if you are using a posbox for instance ^^).
اَلْعَرَبِيَّةُ
is rendered fine in my browser but copying this in sublimetext will switch the order (because it doesn't support RTL).
TL;DR placeholder position doesn't change in RTL languages.
Some languages like Japanese change the order of the words (verb position, etc.) but I don't think it is the case here as it's not really a sentence (but I may be wrong)

Comment (seb-odoo) [reply addons/sale_coupon/models/sale_order.py:105]: There is also the case of punctuation/separator that might be different in some languages.

In this case it depends:
- if the `-` has a semantic meaning in the phrase (in which case it must be translated)
- or if it's just a design separator (in which case it probably should not be a single string in the first place, but 2 values that are not concatenated, and the views having to display them as 2 different information with a separator designed accordingly).

Honestly this seems like it just comes from us lazy (efficient?) developers that wanted to use existing framework possibilities to display the information as easily as possible instead of creating a more correct but more complicated way to pass the additional information.

File: addons/point_of_sale/models/pos_config.py [hunk 2]
```diff
@@ -460,9 +460,9 @@ def name_get(self):
         for config in self:
             last_session = self.env['pos.session'].search([('config_id', '=', config.id)], limit=1)
             if (not last_session) or (last_session.state == 'closed'):
-                result.append((config.id, config.name + ' (' + _('not used') + ')'))
-                continue
-            result.append((config.id, config.name + ' (' + last_session.user_id.name + ')'))
+                result.append((config.id, _("%(pos_name)s (not used)", pos_name=config.name)))
+            else:
+                result.append((config.id, _("%(pos_name)s (%(user)s)", pos_name=config.name, user=last_session.user_id.name)))
```
Comment (mart-e) [inline addons/point_of_sale/models/pos_config.py:465]: remove this one, it doesn't contain anything useful

File: addons/sale/models/sale.py [hunk 3]
```diff
@@ -1229,12 +1229,15 @@ def _update_line_quantity(self, values):
             order_lines = self.filtered(lambda x: x.order_id == order)
             msg = "<b>" + _("The ordered quantity has been updated.") + "</b><ul>"
             for line in order_lines:
-                msg += "<li> %s:" % (line.product_id.display_name,)
-                msg += "<br/>" + _("Ordered Quantity") + ": %s -> %s <br/>" % (
-                line.product_uom_qty, float(values['product_uom_qty']),)
+                msg += "<li> %s: <br/>" % line.product_id.display_name
```
Comment (mart-e) [inline addons/sale/models/sale.py:1232]: What happens if I call my product `</li>` :sweat:

**Comment by Feyensv** [comment]
@robodoo r+
